### PR TITLE
Cleaner example for reduceBy

### PR DESCRIPTION
**Pull request by CrossEye**
The original is pretty hard to follow.

### CODE REVIEW

File: source/reduceBy.js [hunk 1]
```diff
@@ -29,25 +29,22 @@ import _xreduceBy from './internal/_xreduceBy';
...
- *                      {name: 'Bart', score: 62}];
- *      namesByGrade(students);
- *      // {
- *      //   'A': ['Lucy'],
- *      //   'B': ['Drew']
- *      //   // ...,
- *      //   'F': ['Bart']
- *      // }
+ *      const groupNames = (acc, student) => acc.concat(student.name)
+ *      const toGrade = (student, score = student.score) =>
```
Comment (Bradcomp) [inline source/reduceBy.js]: TIL you could define parameters in terms of other parameters in JS. 

I had to run this in the REPL to make sure I knew what was going on, and you have a function that _appears_ to take two parameters even though the sig calls for one.

Would it be simpler to just do a basic destructuring: `const toGrade = ({score}) => ...`?

Comment (CrossEye) [reply source/reduceBy.js]: Yes it would.  I wasn't thinking straight.  I was changing the existing function and wanted to make it a simple expression, and that version was the first thing that came to my mind.  I will to use simple destructuring.
